contrib/cloud.google.com/go/pubsub.v1/pubsub_test.go

```diff
@@ -0,0 +1,218 @@
+		msgID = msg.ID
+		spanID = span.Context().SpanID()
+		traceID = span.Context().TraceID()
+		msg.Ack()
+		called = true
+	}))
+	assert.True(t, called, "callback not called")
+	assert.NoError(t, err)
+
+	spans := mt.FinishedSpans()
+	assert.Len(t, spans, 2, "wrong number of spans")
+	assert.Equal(t, "pubsub.publish", spans[0].OperationName())
+	assert.Equal(t, "pubsub.receive", spans[1].OperationName())
+
+	assert.Equal(t, ext.SpanTypeMessageProducer, spans[0].Tag(ext.SpanType))
+	assert.Equal(t, "projects/project/topics/topic", spans[0].Tag(ext.ResourceName))
+	assert.Equal(t, spans[0].TraceID(), spans[0].SpanID())
+	assert.Equal(t, traceID, spans[0].TraceID())
+	assert.Equal(t, 5, spans[0].Tag("message_size"))
+	assert.Equal(t, 2, spans[0].Tag("num_attributes"))
+	assert.Equal(t, "xxx", spans[0].Tag("ordering_key"))
+	assert.Empty(t, spans[0].Tag("message_id"))
+	assert.Empty(t, spans[0].Tag("publish_time"))
+	assert.Empty(t, spans[0].Tag(ext.Error))
+
+	assert.Equal(t, ext.SpanTypeMessageConsumer, spans[1].Tag(ext.SpanType))
+	assert.Equal(t, "projects/project/subscriptions/subscription", spans[1].Tag(ext.ResourceName))
+	assert.Equal(t, spans[0].SpanID(), spans[1].ParentID())
+	assert.Equal(t, traceID, spans[1].TraceID())
+	assert.Equal(t, spanID, spans[1].SpanID())
+	assert.Equal(t, 5, spans[1].Tag("message_size"))
+	assert.Equal(t, 2, spans[1].Tag("num_attributes"))
+	assert.Equal(t, "xxx", spans[1].Tag("ordering_key"))
+	assert.NotEmpty(t, spans[1].Tag("message_id"))
+	assert.Equal(t, msgID, spans[1].Tag("message_id"))
+	assert.NotEmpty(t, spans[1].Tag("publish_time"))
+	assert.Empty(t, spans[1].Tag(ext.Error))
+}
+
+func TestPropagationNoPubsliherSpan(t *testing.T) {
+	ctx, topic, sub, mt, cleanup := setup(t)
+	defer cleanup()
+
+	// Publisher
+	// no tracing on publisher side
+	_, err := topic.Publish(ctx, &pubsub.Message{Data: []byte("hello"), OrderingKey: "xxx"}).Get(ctx)
+	assert.NoError(t, err)
+
+	// Subscriber
+	var (
+		msgID   string
+		spanID  uint64
+		traceID uint64
+		called  bool
+	)
+	err = sub.Receive(ctx, ReceiveTracer(sub, func(ctx context.Context, msg *pubsub.Message) {
+		assert.False(t, called, "callback called twice")
+		assert.Equal(t, msg.Data, []byte("hello"), "wrong payload")
+		span, ok := tracer.SpanFromContext(ctx)
+		assert.True(t, ok, "no span")
+		msgID = msg.ID
+		spanID = span.Context().SpanID()
+		traceID = span.Context().TraceID()
+		msg.Ack()
+		called = true
+	}))
+	assert.True(t, called, "callback not called")
+	assert.NoError(t, err)
+
+	spans := mt.FinishedSpans()
+	assert.Len(t, spans, 1, "wrong number of spans")
+	assert.Equal(t, "pubsub.receive", spans[0].OperationName())
+
+	assert.Equal(t, ext.SpanTypeMessageConsumer, spans[0].Tag(ext.SpanType))
+	assert.Equal(t, "projects/project/subscriptions/subscription", spans[0].Tag(ext.ResourceName))
+	assert.Equal(t, traceID, spans[0].TraceID())
+	assert.Equal(t, spanID, spans[0].SpanID())
+	assert.Equal(t, 5, spans[0].Tag("message_size"))
```

Comment:
What if we just use `spans[0].Tags()` and compare against a map with a single assert statement?